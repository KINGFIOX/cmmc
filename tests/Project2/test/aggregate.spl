struct empty {};
struct foo {int c;};
struct foo2 {int c = 10;};
int test(int w[1]) {
    w[0] = w[0];
    w = w;
    w[0] = w;
    w = w[0];
    return w[0];
}
int main() {
    struct empty x;
    struct empty y[1];
    struct foo c;
    int z[0];
    int w[1];
    int b;
    
    test(w);
    w[-1] = 10;
    b = w + 10;
    b = w[0] + 10;
    (w[1] + c.c) = w[3];
    b = z + 10;
    b = x + 10;
    x = x + x;
    x = x + b;
    x = b + b;
    y = x;
    y = b;
    y = c;
    (0[w]) = 10;
    test(w) = 10;

    return 0;
}
