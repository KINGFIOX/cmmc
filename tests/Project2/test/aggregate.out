Error type 4294967295 at Line 3: initial values of struct fields are not allowed
Error type 5 at Line 6: unmatching type on both sides of assignment
Error type 5 at Line 7: unmatching type on both sides of assignment
Error type 5 at Line 8: unmatching type on both sides of assignment
Error type 4294967295 at Line 15: invalid array size
Error type 7 at Line 21: binary operation on non-number variables
Error type 5 at Line 21: unmatching type on both sides of assignment
Error type 6 at Line 23: left side in assignment is rvalue
Error type 7 at Line 24: binary operation on non-number variables
Error type 5 at Line 24: unmatching type on both sides of assignment
Error type 7 at Line 25: binary operation on non-number variables
Error type 5 at Line 25: unmatching type on both sides of assignment
Error type 7 at Line 26: binary operation on non-number variables
Error type 5 at Line 26: unmatching type on both sides of assignment
Error type 7 at Line 27: binary operation on non-number variables
Error type 5 at Line 27: unmatching type on both sides of assignment
Error type 5 at Line 28: unmatching type on both sides of assignment
Error type 5 at Line 29: unmatching type on both sides of assignment
Error type 5 at Line 30: unmatching type on both sides of assignment
Error type 5 at Line 31: unmatching type on both sides of assignment
Error type 12 at Line 32: indexing by non-integer
Error type 10 at Line 32: indexing on non-array variable
Error type 5 at Line 32: unmatching type on both sides of assignment
